python/mxnet/contrib/onnx/mx2onnx/_op_translations.py

```diff
@@ -435,6 +435,101 @@ def convert_pad(node, **kwargs):
     return [node]
 
 
+@mx_op.register("dot")
+def convert_dot(node, **kwargs):
+    """Map MXNet's dot operator attributes to onnx's
+    MatMul and Transpose operators based on the values set for
+    transpose_a, transpose_b attributes."""
+    helper, _, _ = import_onnx_modules()
+    proc_nodes = kwargs["proc_nodes"]
+    node_inputs = node["inputs"]
+    name = node["name"]
+
+    input_a_idx = kwargs["index_lookup"][node_inputs[0][0]]
+    input_node_a = proc_nodes[input_a_idx].name
+    input_b_idx = kwargs["index_lookup"][node_inputs[1][0]]
+    input_node_b = proc_nodes[input_b_idx].name
+
+    # Getting the attributes and assigning default values.
+    if "attrs" in node:
+        attrs = node["attrs"]
+        trans_a = int(attrs["transpose_a"])
+        trans_b = int(attrs["transpose_b"])
+    else:
+        trans_a = 0
+        trans_b = 0
+
+    op_name = "transpose" + str(kwargs["idx"])
+
+    if trans_a == 0 and trans_b == 0:
```

Comment:
there are duplicate code block which can be reused by splitting the logic of trans_a and trans_b

Reply:
Simplified